src/Misc/baseError.ts

```diff
@@ -1,12 +1,46 @@
+import { ErrorCodesType } from "./errorCodes";
+
 /**
+ * BabylonJS error internal class wrapper. Extending Error for a class with docs
+ * fails typedoc validation, because lib.es5.d.ts does not have any docs for Error.
  * @ignore
- * Application error to support additional information when loading a file
  */
-export abstract class BaseError extends Error {
+class BabylonJSErrorWrapper extends Error {}
+
+/**
+ * Base Error class for babylonjs errors
+ */
+export class BaseError extends BabylonJSErrorWrapper {
     // See https://stackoverflow.com/questions/12915412/how-do-i-extend-a-host-object-e-g-error-in-typescript
     // and https://github.com/Microsoft/TypeScript/wiki/Breaking-Changes#extending-built-ins-like-error-array-and-map-may-no-longer-work
 
     // Polyfill for Object.setPrototypeOf if necessary.
     protected static _setPrototypeOf: (o: any, proto: object | null) => any =
         (Object as any).setPrototypeOf || ((o, proto) => { o.__proto__ = proto; return o; });
+
+    /**
+     * The error code
+     */
+    public errorCode?: ErrorCodesType;
+
+    /**
+     * The exception that caused the outer exception
+     */
+    public innerException?: Error;

```

Comment:
Adding these to `baseError` is also a bit weird for `LoadFileError` and `RequestFileError` since they both derive from this class but don't have `errorCode` or `innerException`.